src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseResourceAllocationAction.java

```diff
@@ -227,53 +232,48 @@ protected FlowPathPair createFlowPathPair(Flow flow, List<FlowPath> pathsToReuse
             if (pathsToReuseBandwidth != null) {
                 for (FlowPath pathToReuseBandwidth : pathsToReuseBandwidth) {
                     if (pathToReuseBandwidth != null) {
                         for (PathSegment reuseSegment : pathToReuseBandwidth.getSegments()) {
-                            if (pathSegment.getSrcSwitch().equals(reuseSegment.getSrcSwitch())
-                                    && pathSegment.getSrcPort() == reuseSegment.getSrcPort()
-                                    && pathSegment.getDestSwitch().equals(reuseSegment.getDestSwitch())
-                                    && pathSegment.getDestPort() == reuseSegment.getDestPort()) {
-                                allowedOverprovisionedBandwidth += pathToReuseBandwidth.getBandwidth();
-                            }
+                            IslEndpoints isl = new IslEndpoints(reuseSegment.getSrcSwitchId().toString(),
+                                    reuseSegment.getSrcPort(), reuseSegment.getDestSwitchId().toString(),
+                                    reuseSegment.getDestPort());
+                            result.put(isl, result.getOrDefault(isl, 0L) + pathToReuseBandwidth.getBandwidth());
                         }
                     }
                 }
             }
-
-            updateAvailableBandwidth(pathSegment.getSrcSwitchId(), pathSegment.getSrcPort(),
-                    pathSegment.getDestSwitchId(), pathSegment.getDestPort(),
-                    allowedOverprovisionedBandwidth, forceToIgnoreBandwidth);
-        }
-    }
-
-    @VisibleForTesting
-    void updateAvailableBandwidth(SwitchId srcSwitch, int srcPort, SwitchId dstSwitch, int dstPort,
-                                         long allowedOverprovisionedBandwidth,
-                                         boolean forceToIgnoreBandwidth) throws ResourceAllocationException {
-        long usedBandwidth = flowPathRepository.getUsedBandwidthBetweenEndpoints(srcSwitch, srcPort,
-                dstSwitch, dstPort);
-        log.debug("Updating ISL {}_{}-{}_{} with used bandwidth {}", srcSwitch, srcPort, dstSwitch, dstPort,
-                usedBandwidth);
-        long islAvailableBandwidth = islRepository.updateAvailableBandwidth(srcSwitch, srcPort,
-                dstSwitch, dstPort, usedBandwidth);
-        if (!forceToIgnoreBandwidth && (islAvailableBandwidth + allowedOverprovisionedBandwidth) < 0) {
-            throw new ResourceAllocationException(format("ISL %s_%d-%s_%d was overprovisioned",
-                    srcSwitch, srcPort, dstSwitch, dstPort));
+            return result;
+        });
+        // Update ISLs which are occupied by the path.
+        Map<IslEndpoints, Long> updatedIsls = islRepository.updateAvailableBandwidthOnIslsOccupiedByPath(pathId);
+        for (Entry<IslEndpoints, Long> entry : updatedIsls.entrySet()) {
+            IslEndpoints isl = entry.getKey();
+            log.debug("ISL {}_{}-{}_{} was updated with used bandwidth",
```

Comment:
there is no used bw in log

Reply:
Right, the correct log message is in IslRepository. This excessive one has been removed.